File: rectangulo.cpp
 Rectangulo(Punto ii, Punto sd){
        inf_izq = ii;
        sup_der = sd;
    }
    
    bool is_cuadrado(){
        return abs(sup_der.lat - inf_izq.lat) == abs(sup_der.lon - inf_izq.lon);
    }
    
    double calc_area(){
        
        if(is_cuadrado()){
            cout<<"Es cuadrado!"<<endl;
            return pow(inf_izq.calc_dist(sup_der),2)/2; //Solo valido para cuadrados
        }
        return abs(inf_izq.lat - sup_der.lat) * abs(inf_izq.lon - sup_der.lon);
        
    }
    
};

int main()
{
    Punto p;
    Punto p2 = Punto(1,1);
    
    p2.print();
    cout<<endl;
    cout<<p.calc_dist(p2)<<endl;
    
    Rectangulo c = Rectangulo(p,p2);
    
    cout<<c.calc_area()<<endl;
    
    return 0;
}

File: rectangulo.py
class Rectangulo:
    def __init__(self, p0:Punto, p1:Punto, p2:Punto, p3:Punto):
        self.p0 = p0
        self.p1 = p1
        self.p2 = p2
        self.p3 = p3
        self.pts = [self.p0,
                    self.p1,
                    self.p2,
                    self.p3]
    
    def en_rectangulo(self, pt:Punto):
        return self.p0.x < pt.x < self.p1.x and self.p2.y < pt.y < self.p1.y

r = Rectangulo(Punto(0,0), Punto(1,0), Punto(1,-1), Punto(0,-1))
p = Punto(0.5, -1.5)
print(r.en_rectangulo(p))
